recipes/fftw/all/conanfile.py

```diff
@@ -70,64 +87,93 @@ def generate(self):
-            self.cpp_info.components["fftwlib"].system_libs.append("m")
-            if self.options.threads:
-                self.cpp_info.components["fftwlib"].system_libs.append("pthread")
 
         # TODO: to remove in conan v2 once cmake_find_package_* & pkg_config generators removed
         self.cpp_info.filenames["cmake_find_package"] = cmake_config_name
         self.cpp_info.filenames["cmake_find_package_multi"] = cmake_config_name
         self.cpp_info.names["cmake_find_package"] = cmake_namespace
         self.cpp_info.names["cmake_find_package_multi"] = cmake_namespace
-        self.cpp_info.components["fftwlib"].names["cmake_find_package"] = cmake_target_name
-        self.cpp_info.components["fftwlib"].names["cmake_find_package_multi"] = cmake_target_name
-        self.cpp_info.components["fftwlib"].set_property("cmake_target_name", f"{cmake_namespace}::{cmake_target_name}")
-        self.cpp_info.components["fftwlib"].set_property("pkg_config_name", pkgconfig_name)
+
+        for self._current_precision in self._all_precisions:
+            prec_suffix = self._prec_suffix[self._current_precision]
+            cmake_target_name = pkgconfig_name = lib_name = "fftw3" + prec_suffix
+            component_name = f"fftwlib_{self._current_precision}"
+            component = self.cpp_info.components[component_name]
+
+            # TODO: back to global scope in conan v2 once cmake_find_package_* & pkg_config generators removed
+            if self.options.openmp:
+                component.libs.append(lib_name + "_omp")
+            if self.options.threads and not self.options.combinedthreads:
+                component.libs.append(lib_name + "_threads")
+            self.cpp_info.components[component_name].libs.append(lib_name)
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                component.system_libs.append("m")
+                if self._current_precision == QUAD:
+                    component.system_libs.extend(['quadmath'])
+                if self.options.threads:
+                    component.system_libs.append("pthread")
+            self.cpp_info.components[component_name].includedirs.append(os.path.join(self.package_folder, "include"))
+
+            component.names["cmake_find_package"] = cmake_target_name
+            component.names["cmake_find_package_multi"] = cmake_target_name
+            component.set_property("cmake_target_name", f"{cmake_namespace}::{cmake_target_name}")
+            component.set_property("pkg_config_name", pkgconfig_name)
+
+    def package_id(self):
+        if self.info.options.precision != 'deprecated':
+            self.info.options.precision_single = self.info.options.precision == SINGLE
+            self.info.options.precision_double = self.info.options.precision == DOUBLE
+            self.info.options.precision_longdouble = self.info.options.precision == LONGDOUBLE
+            self.info.options.precision_quad = self.info.options.precision == QUAD
```

Comment:
```suggestion
```

Reply:
this will result in weird package IDs, as far as I understand.
the idea is specifying `precision = single` works the same as specifying solely `precision_single=True` and all other precision options to False, and so on. binaries produced in these cases should be identical, therefore, package IDs should be computed identically as well.

Reply:
Keeping two different options doing the same thing does not sounds correct, they will affect the package ID, but result in the same. If we need to deprecate the precision option, than it should be removed from the package ID. It will not result in weird package IDs, but they will follow the new options. Still, people can use lockfiles to avoid breaking.

Reply:
> they will affect the package ID, but result in the same

right now (with my changes) package ID is the same in both situations

>  than it should be removed from the package ID

it's already removed from the package ID, see https://github.com/conan-io/conan-center-index/pull/20888/files#diff-a2da4addc10b5279e914aa8b033195377d13a181faa3a2b3e8e5920796697bd4R170

> It will not result in weird package IDs

with changes your suggests `precision = deprecated` acts as nothing build, means package with no binaries. that's useless package and bad package ID.

Reply:
My point is not using the same package ID, but keeping the recipe as simple as possible. Please, remove this piece that I suggested, it only increases complexity.

Reply:
I can try to remove, let's see, but package produced with `precision != deprecated` will be completely useless (I am not sure if it's okay for consumers).